m4/openssl.m4

```diff
@@ -220,8 +224,8 @@ int main(void) { DTLSv1_method(); }
 AC_MSG_CHECKING([if programs using OpenSSL functions will link])
 AC_LINK_IFELSE(
 	[AC_LANG_SOURCE([[
-#include <openssl/evp.h>
-int main(void) { SSLeay_add_all_algorithms(); }
+#include <openssl/crypto.h>
+int main(void) { return (int) OpenSSL_version_num(); }
```

Comment:
We don't need to use OpenSSL's return value as `main()`'s return value:

```suggestion
int main(void) { OpenSSL_version_num(); return 0; }
```

Reply:
I did explain why above - see https://github.com/uim/uim/pull/198#discussion_r1693976778.

Reply:
Ah, OK. Could you add a comment why we use `OpenSSL_version_num()`'s return value?

Reply:
Done, thanks!